### DISCUSSION TITLE
Start new range selection

### DISCUSSION THREAD

**Discussion by DawnMD** [gpbl/react-day-picker#2050] [Support] [answered]
This library covers most of our use cases except the for eg when I already selected a date range, I want to select a new start date when I click on a new date again but the current implementation is that when a new date is selected and if the date is before the start date in the range the original is moved to the clicked date and same with end date as well. The only way to reset that is to double-click on a specific date which will reset the range and start from new.

Tried playing with the functions `onSelect` and `onDayClick` but not outcomes :( 

Reference to the functionality: Please check the calendar at [booking.com](https://www.booking.com/)

**Comment by severinlandolt** [comment]
Absolutely second this! Maybe a prop `rangeSelectionMode?: "classic" | "extending"`, where classic is what used over at booking.com.

@gpbl Don't know if you knew that, but palantir uses your day picker 😄  Also with an alternative range selection  → https://blueprintjs.com/docs/#datetime2/date-range-picker3

**Comment by gpbl** [comment, marked as answer]
Here is a possible workaround: 

https://github.com/gpbl/react-day-picker/issues/2140#issuecomment-2127651564

See also https://github.com/gpbl/react-day-picker/issues/2140

**Reply by DawnMD** [reply]
After revisiting this, I think there is a issue with the new v9 as with react-hook-form the validation fails as the prop returns a function but not an object

**Reply by gpbl** [reply]
@DawnMD, we’ve made several changes to the range selection mode since then, and I can’t recall the issue you have/had. Could you help me reproduce it by forking this code sandbox? https://codesandbox.io/p/sandbox/daypicker-react-19-2s9csc

Thanks!

**Reply by DawnMD** [reply]
Hi @gpbl I've created on repo to demonstrate the exact issue I'm facing when altering the type of range selection with the mentioned #2140 
Please find it [here](https://github.com/DawnMD/react-day-picker-issue-preview)
Also [preview](https://react-day-picker-issue-preview.vercel.app/)
